opencti-platform/opencti-front/src/private/components/common/stix_core_relationships/StixCoreRelationshipCreationFormStage.tsx

```diff
@@ -0,0 +1,210 @@
+  defaultStopTime,
+  helpers,
+  entityId,
+}) => {
+  const { stixCoreObject } = usePreloadedQuery(
+    stixCoreRelationshipCreationFromEntityQuery,
+    queryRef,
+  );
+
+  const { state: {
+    relationshipTypes: allowedRelationshipTypes,
+    reversed,
+    handleReverseRelation,
+    paginationOptions,
+    connectionKey,
+    onCreate,
+  } } = useContext(CreateRelationshipContext);
+
+  if (!stixCoreObject) {
+    return (
+      <div style={{ display: 'table', height: '100%', width: '100%' }}>
+        <span
+          style={{
+            display: 'table-cell',
+            verticalAlign: 'middle',
+            textAlign: 'center',
+          }}
+        >
+          <CircularProgress size={80} thickness={2} />
+        </span>
+      </div>
+    );
+  }
+
+  const sourceEntity: TargetEntity = stixCoreObject;
+  const [fromEntities, setFromEntities] = useState<TargetEntity[]>([sourceEntity]);
+  const [toEntities, setToEntities] = useState<TargetEntity[]>(targetEntities);
+  useEffect(() => {
+    if (reversed) {
+      setFromEntities(targetEntities);
+      setToEntities([sourceEntity]);
+    } else {
+      setFromEntities([sourceEntity]);
+      setToEntities(targetEntities);
+    }
+  }, [reversed]);
+
+  const onSubmit: FormikConfig<StixCoreRelationshipCreationFromEntityForm>['onSubmit'] = (values, { setSubmitting, setErrors, resetForm }) => {
+    setSubmitting(true);
+    for (const targetEntity of targetEntities) {
+      const fromEntityId = reversed ? targetEntity.id : entityId;
+      const toEntityId = reversed ? entityId : targetEntity.id;
+      const finalValues = {
+        ...values,
+        confidence: parseInt(values.confidence, 10),
+        fromId: fromEntityId,
+        toId: toEntityId,
+        start_time: formatDate(values.start_time),
+        stop_time: formatDate(values.stop_time),
+        killChainPhases: values.killChainPhases.map((kcp) => kcp.value),
+        createdBy: values.createdBy.value,
+        objectMarking: values.objectMarking.map((om) => om.value),
+        externalReferences: values.externalReferences.map((er) => er.value),
+      };
+      try {
+        commitMutation({
+          mutation: reversed
+            ? stixCoreRelationshipCreationFromEntityToMutation
+            : stixCoreRelationshipCreationFromEntityFromMutation,
+          variables: { input: finalValues },
+          updater: (store: RecordSourceSelectorProxy) => {
+            if (connectionKey === 'Pagination_stixCoreRelationships') {
+              // Handles 'Relationships View'
+              insertNode(
+                store,
+                connectionKey,
+                paginationOptions,
+                'stixCoreRelationshipAdd',
+              );
+            } else if (typeof onCreate !== 'function') {
+              // Handle 'Entities View'
+              // TODO: Why is this the only way that properly updates ListLines?
+              const userProxy = store.get(store.getRoot().getDataID());
```

Comment:
you can use insertNode instead of lines 122 to 148

the code is:

insertNode(
store,
connectionKey || 'Pagination_stixCoreRelationships',
paginationOptions,
'stixCoreRelationshipAdd',
null,
null,
null,
reversed ? 'from' : 'to')

Reply:
I have fixed it :)